dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/startup/ConfigurationPopulator.java

```diff
@@ -74,4 +90,20 @@ private void checkSecurityConfiguration() {
       log.info("Encryption is available");
     }
   }
+
+  private static void injectUserInSecurityContext(SystemUser actingUser) {
+    Authentication authentication =
+        new UsernamePasswordAuthenticationToken(actingUser, "", actingUser.getAuthorities());
+    SecurityContext context = SecurityContextHolder.createEmptyContext();
+    context.setAuthentication(authentication);
+    SecurityContextHolder.setContext(context);
+  }
+
+  private static void clearContext() {
+    SecurityContext context = SecurityContextHolder.getContext();
+    if (context != null) {
+      SecurityContextHolder.getContext().setAuthentication(null);
+    }
+    SecurityContextHolder.clearContext();
```

Comment:
Indeed, here is at least one more usage of the same code. I think this is also similar or the same as I added in the `DefaultAuthenticationService` so even that might want to call the same static helper methods.